src/components/ui/DatePicker/DatePickerDropdown.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+
+  export { className as class };
+  let className: string = '';
+  export let value: string | number;
+  export let options: (DropdownOption | DropdownCustomOption)[] = [];
+
+  const dispatch = createEventDispatcher();
+
+  let displayValue: string | number = value;
+
+  $: displayValue = getDisplayValue(value, options);
+
+  function getDisplayValue(value: string | number, options: (DropdownOption | DropdownCustomOption)[]) {
+    if ((options[0] as DropdownCustomOption).value !== undefined) {
+      const option = options.find((option: DropdownCustomOption) => {
+        return option.value === value;
+      });
+      if (option !== undefined) {
+        return (option as DropdownCustomOption).label;
+      }
+    }
+    return value;
+  }
+
+  function getOptionValue(option: DropdownOption | DropdownCustomOption) {
+    if ((option as DropdownCustomOption).value !== undefined) {
+      return (option as DropdownCustomOption).value;
+    }
+
+    return option;
+  }
+
+  function getOptionLabel(option: DropdownOption | DropdownCustomOption) {
+    if ((option as DropdownCustomOption).label !== undefined) {
+      return (option as DropdownCustomOption).label;
+    }
+
+    return option;
+  }
+
+  function onChange(event: Event) {
+    dispatch('change', event);
+  }
+</script>
+
+<div class={`date-picker-dropdown${className ? ` ${className}` : ''}`}>
+  <div class="display-value">
+    <span class="value">{displayValue}</span>
+    <select class="hidden" data-type="number" {value} tabindex="-1" on:change={onChange}>
```

Comment:
So actually we can simplify `on:change={onChange}` to just `on:change` and we can remove the `onChange` function. You should still be able to catch the native DOM change event on `DatePickerDropdown`: https://svelte.dev/tutorial/dom-event-forwarding

Reply:
that's awesome